src/dataflow/src/source/postgres.rs

```diff
@@ -197,26 +197,47 @@ impl PostgresSourceReader {
             .parse()
             .or_else(|_| Err(anyhow!("invalid lsn"))));
         for info in publication_tables {
-            // TODO(petrosagg): use a COPY statement here for more efficient network transfer
-            let data = client
-                .simple_query(&format!(
-                    "SELECT * FROM {:?}.{:?}",
-                    info.namespace, info.name
-                ))
+            let relation_id: Datum = (info.rel_id as i32).into();
+            let reader = client
+                .copy_out_simple(
+                    format!(
+                        "COPY {:?}.{:?} TO STDOUT (FORMAT TEXT, DELIMITER '\t')",
+                        info.namespace, info.name
+                    )
+                    .as_str(),
+                )
                 .await?;
-            for msg in data {
-                if let SimpleQueryMessage::Row(row) = msg {
+
+            pin_mut!(reader);
+            while let Some(Ok(b)) = reader.next().await {
+                // Convert raw rows from COPY into repr:Row. Each Row is a relation_id
+                // and list of string-encoded values, e.g. Row{ 16391 , ["1", "2"] }
+                let mut parser = pgcopy::CopyTextFormatParser::new(b.as_ref(), "\t", "\\N");
+                while !parser.is_eof() && !parser.is_end_of_copy_marker() {
                     let mut mz_row = Row::default();
```

Comment:
You can keep an instance of row outside of this loop and use `finish_and_reuse()` to get owned copies out. If you search for `finish_and_reuse` you'll find many instances of this pattern in our codebase. The intention is to reduce re-allocs as the reused row will keep the maximum size it has ever allocated as scratch space and allocate exact sized ones whenever you call `finish_and_reuse()`

Reply:
I would suggest adding a dedicated test to test/pg-cdc that checks that stuff containing \t and \n can be snapshotted correctly.

Reply:
Switched to `finish_and_reuse` + added some tests with some `\t` and `\n` laden strings